crates/hir-ty/src/tests/macros.rs

```diff
@@ -1373,3 +1373,36 @@ pub fn attr_macro() {}
 "#,
     );
 }
+
+#[test]
+fn clone_with_type_bound() {
+    check_types(
+        r#"
+//- minicore: derive, clone
+#[derive(Clone)]
+struct Float;
+
+trait TensorKind: Clone {
+    /// The primitive type of the tensor.
+    type Primitive: Clone;
+}
+
+impl Clone for f64 {}
```

Comment:
Don't know why this line is required.

Reply:
Not sure, we do have `impl const Clone for f64` in `minicore`.

Reply:
You need to enable `builtin_impls`, that is `//- minicore: derive, clone, builtin_impls`